journal/202303.md

```diff
@@ -0,0 +1,635 @@
+
+{ _Hint: Hints just add context. If it doesn't start with "Hint: ", it is a task. This was a hint._ }
+
+{ _Hint: Read the Guidelines on how to make DJ great: https://github.com/xaur/decred-news/blob/docs/guidelines.md ._}
+
+![{alt text}](../img/{file.ext} "{hover tooltip text}")
+
+_Image: { "Title by @author. Description..." OR just "Description..." }_
+
+{ _Month's highlights, no links here, 5-7 bullets max._
+
+- Bison Relay v0.1.6 RC1
+
+- DCRDEX 0.6 beta testing and release candidates
+
+- Proposal activity
+
+- 
+
+- 
+
+}
+
+Contents: { _Sync ToC at the very end, when all text is finished_ }
+
+- [Development](#development)
+- [People](#people)
+- [Governance](#governance)
+- [Network](#network)
+- [Ecosystem](#ecosystem)
+- [Outreach](#outreach)
+- [Events](#events)
+- [Media](#media)
+- [Markets](#markets)
+- [Relevant External](#relevant-external)
+
+
+## { Some Announcement }
+
+{ _Add individual sections for high-impact announcements, such as consensus votes, software releases, milestones, major bugs, etc._ }
+
+
+## Proposal to change PoW algorithm to BLAKE3 and PoW/PoS split to 1/89
+
+{ it's big so we ann it here briefly }
+
+
+## Bison Relay v0.1.6 Release Candidate
+
+{ highlights, links }
+
+{ cover both v0.1.5 and v0.1.6 rc1 }
+
+
+## Development
+
+The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.
+
+{
+
+- _For each project, cover its most notable dev work._
+- _For each project, the order of reporting is "most released first": changes released to end users, merged to master, in progress (work started), discussions and future plans._
+- _Add sections for any other project with notable activity._
+- _Drop sections without notable updates._
+
+}
+
+
+### dcrd
+
+_[dcrd](https://github.com/decred/dcrd) is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+Hardening of consensus change voting code continued from [last month](202302.md#dcrd):
+
+- Enhanced startup [validation](https://github.com/decred/dcrd/pull/3068) of deployment parameters to ensure they satisfy the assumptions that the vote tallying logic rely on. For example, it checks that each vote has exactly one Abstain and one No choice, that there are no duplicate vote choices, or that choices are properly encoded in bits. This may unlock future optimizations and simplifications of code related to vote tallying.
+  - { _very optional for general understanding: (1) How validation allows to optimize vote tallying? (2) These changes are not in the consensus layer right? i.e. would blocks failing these new checks be rejected by some other code?_ }
```

Comment:
1. It means the validation only needs to be done once instead of every single time agenda results are determined.
2. Correct.  They're not consensus.  At least not directly as you're asking.  Rather, it's effectively compile/initialization time assertions to ensure the underlying parameters used in the consensus code don't violate any assumptions the consensus code makes on them.  Failing these checks means `dcrd` would not even launch, so there isn't any chance of a consensus break between versions due to these.

Reply:
Thanks, this is good to know. I added a simplified version of that:

> Note that the above changes are not breaking the consensus between versions in any way, they add more checks that allow dcrd to fail earlier if it detects that any assumptions made by the consensus code are violated.